control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go

```diff
@@ -0,0 +1,235 @@
+package oapi
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Ensure certain deployment fields do not get set
+func TestReconcileOpenshiftAPIServerDeploymentNoChanges(t *testing.T) {
```

Comment:
please follow convention for naming TestFuncName, i.e. TestReconcileDeployment()
then no changes should be a test case.

Reply:
The tests is already in that way in main, I didn't created this. The new one I've included in the PR is `TestReconcileOpenshiftAPIServerDeploymentTrustBundle` the other was in the file already and I though that was ok, to keep it in the backport.